examples/cylinders.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2015 The Noise-rs Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate noise;
+extern crate image;
+extern crate num;
+
+use noise::*;
+use self::num::{Float, NumCast, Zero};
+use std::path::Path;
+use noise::modules::Cylinders;
+
+fn main() {
+    render_cylinders::<f64>("cylinders.png", 1.0, 1024, 1024, 50);
+    println!("Generated cylinders.png");
+}
+
+fn cast<T: NumCast, R: NumCast>(val: T) -> R {
+    num::traits::cast(val).unwrap()
+}
+
+fn clamp<F: Float>(val: F, min: F, max: F) -> F {
+    match () {
+        _ if val < min => min,
+        _ if val > max => max,
+        _ => val,
+    }
+}
+
+fn render_cylinders<T>(filename: &str, frequency: T, width: u32, height: u32, zoom: usize)
+    where T: Float + NumCast
+{
+    let mut pixels = Vec::with_capacity((width * height) as usize);
+
+    let cyls = Cylinders::new(frequency);
+
+    for y in 0..height {
+        for x in 0..width {
+            let value: f64 =
+                cast(cyls.get([(cast::<_, T>(x) - cast::<_, T>(width / 2)) / cast::<_, T>(zoom),
+                               T::zero(),
+                               (cast::<_, T>(y) - cast::<_, T>(height / 2)) / cast::<_, T>(zoom)]));
+            pixels.push(cast(clamp(value * 0.5 + 0.5, 0.0, 1.0) * 255.0));
+        }
+    }
+
+    let _ = image::save_buffer(&Path::new(filename),
+                               &*pixels,
+                               width,
+                               height,
+                               image::Gray(8));
+}
```

Comment:
The examples probably don't need to be generic - `f32` or `f64` should be fine! :)